### PR TITLE
[JBPM-8487] Generate frontend tests results in junit format

### PR DESCRIPTION
**Pull request by ruromero**
Fix https://issues.jboss.org/browse/JBPM-8487

Signed-off-by: Ruben Romero Montes <rromerom@redhat.com>

### CODE REVIEW

**Comment by kie-ci2** [comment]
Can one of the admins verify this PR? Comment with 'ok to test' to start the build.

File: process-migration-service/frontend/package.json [hunk 1]
```diff
@@ -16,13 +20,12 @@
...
-    "babel",
-    "minimal",
-    "starter"
-  ],
+  "jest-junit": {
+    "outputDirectory": "../target",
+    "suiteName": "org.kiegroup.processmigration.js.tests",
+    "suiteNameTemplate": "{filename}",
+    "classNameTemplate": "org.kiegroup.processmigration.js.tests.{filename}.{classname}"
+  },
```
Comment (MarianMacik) [inline process-migration-service/frontend/package.json]: Can we maybe put this to jest.config.js so it is not scattered over more files?

I have added localy this:

```
                    "jest-junit",
                    {...
                     titleTemplate:...
                     ...
                    "outputDirectory": "../target/jest-junit-reports",
                    "outputName":"TEST-org.kiegroup.processmigration.js.tests.xml",
                    "suiteName": "org.kiegroup.processmigration.js.tests",
                    "suiteNameTemplate": "{filename}",
                    "classNameTemplate": "org.kiegroup.processmigration.js.tests.{filename}.{classname}"
```

Notice also the name and location of the resulting xml file. On Jenkins we filter these jUnit reports by `**/target/*-reports/TEST-*.xml`. I have also let @domhanak know that we should unify it and not leave the default `junit.xml`.

File: process-migration-service/frontend/package-lock.json [hunk 2]
```diff
@@ -1958,7 +1963,8 @@
     "@types/d3-time": {
       "version": "1.0.10",
       "resolved": "https://registry.npmjs.org/@types/d3-time/-/d3-time-1.0.10.tgz",
-      "integrity": "sha512-aKf62rRQafDQmSiv1NylKhIMmznsjRN+MnXRXTqHoqm0U/UZzVpdrtRnSIfdiLS616OuC1soYeX1dBg2n1u8Xw=="
+      "integrity": "sha512-aKf62rRQafDQmSiv1NylKhIMmznsjRN+MnXRXTqHoqm0U/UZzVpdrtRnSIfdiLS616OuC1soYeX1dBg2n1u8Xw==",
+      "optional": true
```
Comment (MarianMacik) [inline process-migration-service/frontend/package-lock.json]: Any idea why is that suddenly optional?

Comment (ruromero) [reply process-migration-service/frontend/package-lock.json]: Nope, npm install stuff

Comment (ruromero) [reply process-migration-service/frontend/package-lock.json]: It went away now. I don't know why it appeared or why it's not there anymore ¯\_(ツ)_/¯

File: process-migration-service/frontend/package.json (hunk 1 above)
Comment (ruromero) [reply process-migration-service/frontend/package.json]: Done

**Comment by MarianMacik** [comment]
ok to test

**Comment by MarianMacik** [comment]
jenkins retest this

**Comment by MarianMacik** [comment]
jenkins retest this

**Comment by MarianMacik** [comment]
There's again the `org.kie.server.integrationtests.controller.WebSocketKieControllerNotificationIntegrationTest.testKieServerEvents` failed test.

```
org.mockito.exceptions.verification.TooManyActualInvocations: 

eventHandler.onServerTemplateUpdated(<any>);
Wanted 1 time:
-> at org.kie.server.integrationtests.controller.WebSocketKieControllerNotificationIntegrationTest.testKieServerEvents(WebSocketKieControllerNotificationIntegrationTest.java:154)
But was 2 times. Undesired invocation:
-> at org.kie.server.controller.client.websocket.WebSocketEventHandler.accept(WebSocketEventHandler.java:50)

	at org.kie.server.integrationtests.controller.WebSocketKieControllerNotificationIntegrationTest.testKieServerEvents(WebSocketKieControllerNotificationIntegrationTest.java:154)
```

Will try to run it once more.

**Comment by MarianMacik** [comment]
jenkins retest this

**Pull request merged**